### PR TITLE
[Merged by Bors] - chore: fix non-class instances in factorization

### PR DESCRIPTION
**Pull request by alexjbest**
See https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/isClass.3F.20panic!/near/391792347

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Category/Factorisation.lean [hunk 1]
```diff
@@ -112,11 +112,14 @@ instance : Unique (d ⟶ (Factorisation.terminal : Factorisation f)) where
...
-instance : IsInitial (Factorisation.initial : Factorisation f) := IsInitial.ofUnique _
+/-- The initial factorization is an initial object -/
+def IsInitial_initial : IsInitial (Factorisation.initial : Factorisation f) := IsInitial.ofUnique _
 
 instance : HasInitial (Factorisation f) := Limits.hasInitial_of_unique Factorisation.initial
 
-instance : IsTerminal (Factorisation.terminal : Factorisation f) := IsTerminal.ofUnique _
+/-- The terminal factorization is a terminal object -/
+def IsTerminal_terminal : IsTerminal (Factorisation.terminal : Factorisation f) :=
+IsTerminal.ofUnique _
```
Comment (alreadydone) [inline Mathlib/CategoryTheory/Category/Factorisation.lean:122]: Hmm, rare non-american spelling of Factorisation in this file (including file name).

**Review by alreadydone — APPROVED** [review]

File: Mathlib/CategoryTheory/Category/Factorisation.lean [hunk 2]
```diff
@@ -112,11 +112,14 @@ instance : Unique (d ⟶ (Factorisation.terminal : Factorisation f)) where
...
 open Limits
 
-instance : IsInitial (Factorisation.initial : Factorisation f) := IsInitial.ofUnique _
+/-- The initial factorization is an initial object -/
+def IsInitial_initial : IsInitial (Factorisation.initial : Factorisation f) := IsInitial.ofUnique _
 
 instance : HasInitial (Factorisation f) := Limits.hasInitial_of_unique Factorisation.initial
 
-instance : IsTerminal (Factorisation.terminal : Factorisation f) := IsTerminal.ofUnique _
+/-- The terminal factorization is a terminal object -/
```
Comment (alexjbest) [inline Mathlib/CategoryTheory/Category/Factorisation.lean:120]:
```suggestion
/-- The terminal factorisation is a terminal object -/
```

File: Mathlib/CategoryTheory/Category/Factorisation.lean [hunk 3]
```diff
@@ -112,11 +112,14 @@ instance : Unique (d ⟶ (Factorisation.terminal : Factorisation f)) where
 
 open Limits
 
-instance : IsInitial (Factorisation.initial : Factorisation f) := IsInitial.ofUnique _
+/-- The initial factorization is an initial object -/
```
Comment (alexjbest) [inline Mathlib/CategoryTheory/Category/Factorisation.lean:115]:
```suggestion
/-- The initial factorisation is an initial object -/
```

File: Mathlib/CategoryTheory/Category/Factorisation.lean (hunk 1 above)
Comment (alexjbest) [reply Mathlib/CategoryTheory/Category/Factorisation.lean]: I fixed my comments to conform. Apparently despite being from the UK I have become americanis(z)ed.

**Comment by semorrison** [comment]
bors merge

File: Mathlib/CategoryTheory/Category/Factorisation.lean (hunk 1 above)
Comment (javra) [reply Mathlib/CategoryTheory/Category/Factorisation.lean]: Yes, there's some alignment to be done at some point, it's called "factorisation" in the category theory part of mathlib and "factorization" everywhere else.

**Pull request closed without merging**